### DISCUSSION TITLE
Size of the unique shapes to help the localization

### DISCUSSION THREAD

**Discussion by abylikhsanov** [stella-cv/stella_vslam#226] [Q&A] [answered]
Hi,

While mapping a big area with more than 50% of FoV being a ceiling (for unique features), I am keen to add some unique printouts/shapes on the ceiling to help the localization system to localize in a more robust way.

If I would fo with something like QR codes, what do you think the size in percentage of the total FoV should be good enough to localize against it? As I understand correctly, if there are a lot of features, this unique prinout should be of a significat size to dominate the decision.

**Comment by ymd-stella** [comment, marked as answer]
Are you considering the method before you define the problem? If you can place markers on the ceiling, then you can simply localize with just the markers. There is no need to use OpenVSLAM.
